src/app/services/login-service/login-service.service.ts

```diff
@@ -0,0 +1,35 @@
+import { LoginResponse } from './../../models/login';
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
+import { Observable} from 'rxjs';
+import { environment } from 'src/environments/environment';
+import { map } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class LoginService {
+
+  constructor(private http: HttpClient) { }
+  login(username, password): Observable<LoginResponse> {
+    const headers = new HttpHeaders({
+      'Content-Type': 'application/json',
+      'Access-Control-Allow-Origin': '*'
+    });
+    const options = {
+      headers: headers,
+      observe: 'response' as 'body'
+    };
+
+    const request = this.http.post(`${environment.hostname}/login`,
+      {
+        'username': `${username}`,
+        'password': `${password}`,
+      }, options);
+
+      return request.pipe(map(res => {
+        const responseData = new LoginResponse(res['body'].userId, res['status']);
+        return responseData;
+      }));
```

Comment:
can replace with this:
`return request.pipe(map(res => new LoginResponse(res['body'].userId, res['status'])));`

Reply:
Nice one, thanks!